ASAA-36 - Use Firebase Remote Config to Check Feature Toggle Flags

jacob-taggart opened the pull request:
[ASAA-36](https://jira.bottlerocketapps.com/browse/ASAA-36)

**Targeting the 'feature/ASA-39-FeatureToggleJSON' branch as it was easier to test on using the created functionality for that.** The original PR comments for ASAA-39 have been addressed. This branch has been updated to include those changes as well as add string/enum implementation from remote config. Once we merge ASAA-39 branch, we can re-target this to the overall feature's branch.

- Build Firebase Remote Config Implementation
- Define and pass in some data class inputs that contain setup/configuration data needed at runtime
- Ensure the implementation is pluggable/replaceable with another implementation that adheres to a common feature toggle interface/contract

Each time the app gets restarted, the remote config values will be checked. If the minimumFetchInterval has been reached, the app will check the remote config. If minimumFetchInterval has not been reached since the previous app start, the app will revert to the previous fetched value. 

To use the in-app defaults (xml) for a certain parameter, check the bottom right option in below screenshot via the Firebase Remote Config console:

<img width="600" alt="Screenshot 2022-12-28 at 3 56 17 PM" src="https://user-images.githubusercontent.com/93532693/209871051-dca9eaeb-5933-4c0b-8190-89b4f6eb7f9a.png">

For more involved config, we can use different data types in the remote config such as String, Number, Boolean, and JSON. The JSON data type would allow for more complex configuration of feature toggling for each feature (i.e. having a boolean param indicating sub-feature functionality implementation or non-implementation which would be denoted by other params, etc.).

**It's possible we may want to change the Remote Config params to be JSONs so that we can have a value for whether or not we need runtime restart such as we are aiming for in the other feature toggle implementations.** This may be desirable for instances in which the app initialization itself requires different processes, as the config check happens as it launches. The chances of this are likely low, but it's something to keep in mind.

jacob-taggart committed:
ASAA-36 - Set up Firebase Remote Config in app and use to fetch Feature Toggle values

LuisRocketRangel reviewed (approved):
Looks great, good job!

jacob-taggart committed:
Merge branch 'feature/ASAA-39-FeatureToggleJSON' into feature/ASAA-36-FeatureToggle-FirebaseRemoteConfig

jacob-taggart committed:
ASAA-36 - change to include string/enum config possibility for Firebase Remote Config implementation

jacob-taggart committed:
ASAA-36 - change remote_config_defaults to include new remote config value

jacob-taggart committed:
ASAA-36 - retool to use FeatureToggle class type

jacob-taggart committed:
ASAA-36 - small formatting change for readability

jacob-taggart committed:
ASAA-36 - change remote config UI flow values to use the newly implemented contract

jacob-taggart committed:
ASAA-36 - minor fix

jacob-taggart committed:
ASAA-36 - readability
